frontend/components/ManageStakingPage/StakingContractSection/index.tsx

```diff
@@ -225,30 +284,67 @@ export const StakingContractSection = ({
     if (activeStakingProgram === stakingProgram)
       return StakingProgramStatus.Selected;
 
-    // not staked, set the default as the Selected contract
-    if (stakingProgram === defaultStakingProgram)
+    // Pearl is not staked, set as Selected if default
+    if (
+      activeStakingProgram === null &&
+      stakingProgram === defaultStakingProgram
+    )
       return StakingProgramStatus.Selected;
 
-    // otherwise, no tag
+    // Otherwise, no tag
     return;
   }, [activeStakingProgram, defaultStakingProgram, stakingProgram]);
 
-  // Show funding address
-  const canShowFundingAddress =
-    !isSelected &&
-    activeStakingContractSupportsMigration &&
-    !hasEnoughOlasToMigrate;
+  /** Displays the dropdown function section & button */
+  const isFundingSectionShown = useMemo(() => {
+    if (!isBalanceLoaded || !isStakingContractInfoLoaded) {
+      return false;
+    }
+
+    if (!isSelectedOrUnstakedDefault) {
+      return false;
+    }
+
+    if (!hasEnoughOlasToMigrate) {
+      return false;
+    }
+
+    // check if migration is possible according to meta
+    // ignore `null` as unstaked
+    if (
+      activeStakingProgram !== null &&
```

Comment:
should this include `isServiceStakedForMinimumDuration` and `!isServiceStaked` logic?